src/WebJobs.Extensions.DurableTask/EventSourceListener.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class EventSourceListener : EventListener
+    {
+        private readonly LinuxAppServiceILogger logger;
+
+        public EventSourceListener(LinuxAppServiceILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            if (eventSource.Name == "DurableTask-Core" ||
+                eventSource.Name == "DurableTask-AzureStorage" ||
+                eventSource.Name == "WebJobs-Extensions-DurableTask")
+            {
+                this.EnableEvents(eventSource, EventLevel.LogAlways, EventKeywords.All);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            this.logger.Log(LogLevel.Information, eventData.EventId, eventData, null, null);
```

Comment:
@davidmrdavid you're gonna hate me for this (if you don't already by now 😅), but now that we're using `EventListener` instead of following my original `ILogger` vision (which was flawed - I think you're going in a much more sustainable direction), I suggest we not use `ILogger` at all and instead just inline the logic that you wrote in the `LinuxAppServiceILogger`/`LinuxDedicatedLogger` code (it's up to you if you want to maintain an abstraction that differentiates between Linux dedicated and Linux Consumption, but I would suggest an abstraction that is *not* ILogger). Hopefully you'll agree with me that this will result in a much simpler solution with far less boilerplate.